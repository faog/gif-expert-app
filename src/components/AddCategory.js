import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (evt) => {
    setInputValue(evt.target.value);
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories(categories => [inputValue, ...categories]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <p>{ inputValue }</p>
      <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
};
