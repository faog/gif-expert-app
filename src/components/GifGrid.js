import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async() => {
    const url = 'https://api.giphy.com/v1/gifs/search?api_key=SehYrvVMKCk9VRuh9nyEK2w66JLEITpL&q=onepunchman&limit=10';
    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map(img => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url
      };
    });

    setImages(gifs);
  };

  return (
    <div>
      <h3>{ category }</h3>
      {
        images.map(img =>
          <GifGridItem
            key={img.id}
            {...img}
          />
        )
      }
    </div>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string
};
