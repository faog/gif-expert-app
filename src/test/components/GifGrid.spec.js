import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import '@testing-library/jest-dom';
jest.mock('../../hooks/useFetchGifs');

describe('<GifGrid /> Component', () => {
  const category = 'Manzana';

  test('should be show component', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should be show items when useFetchGifs load ', () => {
    const gifs = [{
      id: 'ABC',
      title: 'Manzana',
      url: 'https://localhost/Manzana.jpg'
    }, {
      id: '123',
      title: 'Naranja',
      url: 'https://localhost/Naranja.jpg'
    }];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
