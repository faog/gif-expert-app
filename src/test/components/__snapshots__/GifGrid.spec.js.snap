// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid /> Component should be show component 1`] = `
<Fragment>
  <h3>
    Manzana
  </h3>
  <p>
    loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`<GifGrid /> Component should be show items when useFetchGifs load  1`] = `
<Fragment>
  <h3>
    Manzana
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Manzana"
      url="https://localhost/Manzana.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Naranja"
      url="https://localhost/Naranja.jpg"
    />
  </div>
</Fragment>
`;
